lib/galaxy/authnz/managers.py

```diff
@@ -100,8 +100,7 @@ def _parse_oidc_backends_config(self, config_file):
     def _parse_idp_config(self, config_xml):
         rtv = {
             'client_id': config_xml.find('client_id').text,
-            'client_secret': config_xml.find('client_secret').text,
-            'redirect_uri': config_xml.find('redirect_uri').text}
```

Comment:
Are we positive the on-the-fly URL generation will never fail? - proxies, mirrors etc. Because here we are removing the option for admins to configure it.

Reply:
Also are we confident that Admin will use the same `redirect_uri` for IdP as the one we will generate?

Reply:
I have not tested `trans.request.host_url` with proxies or mirrors. If anyone have proxy/mirror setup in place, I would appreciate if this they can test this. Otherwise, I can set up proxy and mirror.